gems/smithy/lib/smithy/vise/resource_shape.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Vise
+    # Represents a resource shape from the model.
+    class ResourceShape < Shape
```

Comment:
I like that we have dedicated classes for `ResourceShape` and such. It makes things easier to read (and properly maintain down the road).